nomos-da/network/core/src/protocols/dispersal/validator/behaviour.rs

```diff
@@ -145,3 +145,431 @@ impl<M: MembershipHandler<Id = PeerId, NetworkId = SubnetworkId> + 'static> Netw
+            })
+            .unwrap()
+            .with_swarm_config(|cfg| {
+                cfg.with_idle_connection_timeout(std::time::Duration::from_secs(u64::MAX))
+            })
+            .build()
+    }
+
+    pub fn validator_swarm(
+        key: Keypair,
+        membership: impl MembershipHandler<NetworkId = u32, Id = PeerId> + 'static,
+    ) -> libp2p::Swarm<
+        DispersalValidatorBehaviour<impl MembershipHandler<NetworkId = u32, Id = PeerId>>,
+    > {
+        libp2p::SwarmBuilder::with_existing_identity(key)
+            .with_tokio()
+            .with_other_transport(|keypair| quic::tokio::Transport::new(quic::Config::new(keypair)))
+            .unwrap()
+            .with_behaviour(|_key| DispersalValidatorBehaviour::new(membership))
+            .unwrap()
+            .with_swarm_config(|cfg| {
+                cfg.with_idle_connection_timeout(std::time::Duration::from_secs(u64::MAX))
+            })
+            .build()
+    }
+
+    fn prepare_swarm_config(
+        num_instances: usize,
+        group_id: usize,
+    ) -> Vec<(Keypair, PeerId, Multiaddr)> {
+        let mut configs = Vec::with_capacity(num_instances);
+
+        for i in 0..num_instances {
+            let keypair = identity::Keypair::generate_ed25519();
+            let peer_id = PeerId::from(keypair.public());
+            let port = 5200 + group_id * num_instances + i;
+            let addr: Multiaddr = format!("/ip4/127.0.0.1/udp/{port}/quic-v1")
+                .parse()
+                .unwrap();
+
+            configs.push((keypair, peer_id, addr));
+        }
+        configs
+    }
+
+    fn create_membership(
+        num_instances: usize,
+        subnet_id: u32,
+        peer_ids: &[PeerId],
+    ) -> Neighbourhood {
+        let mut membership = HashMap::default();
+
+        for peer_id in peer_ids.iter().take(num_instances) {
+            membership.insert(*peer_id, HashSet::from([subnet_id]));
+        }
+
+        Neighbourhood { membership }
+    }
+
+    async fn run_executor_swarm(
+        mut swarm: Swarm<
+            DispersalExecutorBehaviour<
+                impl MembershipHandler<NetworkId = u32, Id = PeerId> + Sized + 'static,
+            >,
+        >,
+        messages_to_expect: usize,
+        mut terminator_rx: watch::Receiver<bool>,
+    ) -> usize {
+        let mut msg_counter = 0;
+        loop {
+            tokio::select! {
+                Some(event) = swarm.next() => {
+                    debug!("Executor event: {event:?}");
+                    if let SwarmEvent::Behaviour(DispersalExecutorEvent::DispersalSuccess{..}) = event {
+                        msg_counter += 1;
+                    }
+                    if msg_counter >= messages_to_expect {
+                        break;
+                    }
+                }
+
+                _ = terminator_rx.changed() => {
+                    if *terminator_rx.borrow() {
```

Comment:
Since we are not using the value, the if condition can be removed, the `break` needs to happen if terminator_rx changes.

Reply:
if condition removed at https://github.com/logos-co/nomos-node/pull/819/commits/e7f90a6bc64c6f0edb1a6fc250df98394ee19364